streams/src/main/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdater.java

```diff
@@ -496,6 +504,103 @@ private void addTask(final Task task) {
             }
         }
 
+        private void removeTask(final TaskId taskId, final CompletableFuture<RemovedTaskResult> future) {
+            try {
+                if (updatingTasks.containsKey(taskId)) {
+                    removeUpdatingTask(taskId, future);
```

Comment:
Could all four `remove...Task` methods just return a boolean if successful?

Reply:
I will try that.

Reply:
Done! Code looks better now. Thanks a lot!